docs/articles/workflow.md

```diff
@@ -0,0 +1,289 @@
+  },
+  "http": {
+      "address": "$ZOT_HOST",
+      "port": "$ZOT_PORT"
+  },
+  "log": {
+      "level": "error"
+  }
+}
+EOF
+  # start zot as a background task
+  ${ZOT} serve $TEST_TMPDIR/zot-config.json &
+  pid=$!
+        # wait until service is up
+  count=5
+  up=0
+  while [[ $count -gt 0 ]]; do
+    if [ ! -d /proc/$pid ]; then
+      echo "zot failed to start or died"
+      exit 1
+    fi
+    up=1
+    curl -f http://$ZOT_HOST:$ZOT_PORT/v2/ || up=0
+    if [ $up -eq 1 ]; then break; fi
+    sleep 1
+    count=$((count - 1))
+  done
+  if [ $up -eq 0 ]; then
+    echo "Timed out waiting for zot"
+    exit 1
+  fi
+  # setup an OCI client
+  ${REGCLIENT} registry set --tls=disabled $ZOT_HOST:$ZOT_PORT
+}
+
+# function for stopping zot after demonstration
+function zot_teardown() {
+  killall zot
+}
+
+# call the function to start zot
+zot_setup
+{% endraw %}
+```
+
+
+### Step 3: Copy an image to the zot registry
+
+This step copies a busybox container image into the zot registry.
+
+```shell
+{% raw %}
+skopeo copy --format=oci --dest-tls-verify=false docker://busybox:latest docker://${ZOT_HOST}:${ZOT_PORT}/busybox:latest
+{% endraw %}
+```
+
+
+### Step 4: Copy an artifact to the zot registry
+
+This step creates a simple artifact file and associates it with the busybox image in the zot registry.
+
+```shell
+{% raw %}
+cat > ${TEST_TMPDIR}/artifact.yaml << EOF
+key:
+  val: artifact
+EOF
+${REGCLIENT} artifact put --artifact-type application/yaml -f ${TEST_TMPDIR}/artifact.yaml --subject ${ZOT_HOST}:${ZOT_PORT}/busybox:latest
+{% endraw %}
+```
+
+> :pencil2: If no subject is specified in the command, the artifact is considered independent and not associated with any existing image.
+
+
+### Step 5: Display the artifact tree
+
+This step prints the artifact tree, including the yaml artifact. The tree shows the association of the artifact to the busybox image.
+
+```shell
+{% raw %}
+REF0=$(${REGCLIENT} artifact tree --format '{{jsonPretty .}}' localhost:8080/busybox:latest | jq .referrer[0].reference.Digest)
+REF0="${REF0:1:-1}"
+{% endraw %}
+```
+
+
+### Step 6: Sign the image and artifact
+
+This step creates a key pair for cosign in a separate directory, then uses cosign to sign the image and artifact. Both signatures, like the artifact file itself, are associated with the busybox image. 
+
+```shell
+{% raw %}
+# create a key pair in a different directory
+pushd ${TEST_TMPDIR}
+COSIGN_PASSWORD= ${COSIGN} generate-key-pair
+popd
+# sign the image
+COSIGN_PASSWORD= COSIGN_OCI_EXPERIMENTAL=1 COSIGN_EXPERIMENTAL=1 ${COSIGN} sign -y --key ${TEST_TMPDIR}/cosign.key --registry-referrers-mode=oci-1-1 ${ZOT_HOST}:${ZOT_PORT}/busybox:latest
+# sign the artifact referring to the image
+COSIGN_PASSWORD= COSIGN_OCI_EXPERIMENTAL=1 COSIGN_EXPERIMENTAL=1 ${COSIGN} sign -y --key ${TEST_TMPDIR}/cosign.key --registry-referrers-mode=oci-1-1 ${ZOT_HOST}:${ZOT_PORT}/busybox@${REF0}
+{% endraw %}
+```
+
+
+### Step 7: Display the artifact tree
```

Comment:
```+ $ regctl artifact tree localhost:8080/busybox:latest
Ref: localhost:8080/busybox:latest
Digest: sha256:9172c5f692f2c65e4f773448503b21dba2de6454bd159905c4bf6d83176e4ea3
Referrers:
  - sha256:9c0655368b10ca4b2ffe39e4dd261fb89df25a46ae92d6eb4e6e1792a451883e: application/yaml
    Referrers:
      - sha256:06792b209137486442a2b804b2225c0014e3e238d363cdbea088bbd73207fb34: application/vnd.dev.cosign.artifact.sig.v1+json
  - sha256:995b6a78bf04a7a9676dac76b4598ccb645c17e30b02f294de9fdfa2f28eb7b2: application/vnd.dev.cosign.artifact.sig.v1+json
```

Reply:
original image: localhost:8080/busybox:latest
Two direct refererrers:
artifact.yaml sha256:9c0655368b10ca4b2ffe39e4dd261fb89df25a46ae92d6eb4e6e1792a451883e: application/yaml
image signature sha256:995b6a78bf04a7a9676dac76b4598ccb645c17e30b02f294de9fdfa2f28eb7b2: application/vnd.dev.cosign.artifact.sig.v1+json

Second-level referrer for the artifact.yaml
- sha256:06792b209137486442a2b804b2225c0014e3e238d363cdbea088bbd73207fb34: application/vnd.dev.cosign.artifact.sig.v1+json

Reply:
Command output and explanation added